### PR TITLE
feat: set post curl command after meteor ran successfully

### PR DESCRIPTION
**Pull request by luthfifahlevi**
**Goal:**
Hit Delete Assets by Query Expr API from Compass, to delete stales assets. The config will be imported from Terraform.

### CODE REVIEW

File: stable/meteor/templates/cronjob.yaml [hunk 1]
```diff
@@ -33,6 +33,20 @@ spec:
               - -c
               - |
                 meteor run /opt/recipes;
+                METEOR_EXIT_CODE=$?
+                echo "Meteor command exited with code METEOR_EXIT_CODE"
```
Comment (irainia) [inline stable/meteor/templates/cronjob.yaml:37]: do we intend to produce the actual code value here?

File: stable/meteor/templates/cronjob.yaml [hunk 2]
```diff
@@ -33,6 +33,20 @@ spec:
...
+
+                if [ $METEOR_EXIT_CODE -eq 0 ]; then
+                  {{- if .Values.cleanupStaleAssets.enabled }}
```
Comment (irainia) [inline stable/meteor/templates/cronjob.yaml:40]: does it mean we will not do anything (not even writing message) when cleanup is not enabled?

Comment (irainia) [inline stable/meteor/templates/cronjob.yaml:40]: indentation may need to be rechecked

**Review by irainia — CHANGES_REQUESTED** [review]
Please update the chart version as well

File: stable/meteor/templates/cronjob.yaml (hunk 2 above)
Comment (luthfifahlevi) [reply stable/meteor/templates/cronjob.yaml:40]: yap, I create it so it is more flexible and easier when we need to disable the auto clean up assets for a certain reason

File: stable/meteor/templates/cronjob.yaml (hunk 1 above)
Comment (luthfifahlevi) [reply stable/meteor/templates/cronjob.yaml:37]: hmm, actually not sure too haha, wdyt mas? I simply thought it's easier to see why the auto clean up not running if it happened. But I just remember we also can see more detail about it in meteor logs hehe

Comment (luthfifahlevi) [reply stable/meteor/templates/cronjob.yaml:37]: I changed my mind, it is no need to echo there hehe thank you mas

File: stable/meteor/templates/cronjob.yaml (hunk 2 above)
Comment (irainia) [reply stable/meteor/templates/cronjob.yaml]: it was just confirmation, whether we want to add message or something here. if it is expected to have nothing to do, then all good.

**Review by irainia — APPROVED** [review]

File: stable/meteor/values.yaml [hunk 3]
```diff
@@ -38,6 +38,11 @@ resources:
 #         foo: "bar"
 #     sinks:
 #       - name: console
+cleanup:
+  enabled: false
+  image: curlimages/curl:8.9.1
```
Comment (haveiss) [inline stable/meteor/values.yaml:43]: where this image will be used?

Comment (luthfifahlevi) [reply stable/meteor/values.yaml:43]: Not used mas. Will delete it 👍  thank you

File: stable/meteor/templates/cronjob.yaml [hunk 4]
```diff
@@ -25,111 +25,118 @@ spec:
...
+              imagePullPolicy: {{ .Values.image.pullPolicy }}
+              command:
+                - /bin/sh
+                - -c
+                - |
+                  meteor run /opt/recipes;
+                  METEOR_EXIT_CODE=$?
+
+                  if [ $METEOR_EXIT_CODE -eq 0 ]; then
+                    {{- if .Values.cleanup.enabled }}
```
Comment (haveiss) [inline stable/meteor/templates/cronjob.yaml:39]: let's make it more generic, like postCmd or something

File: stable/meteor/templates/cronjob.yaml [hunk 5]
```diff
@@ -25,111 +25,118 @@ spec:
...
+              command:
+                - /bin/sh
+                - -c
+                - |
+                  meteor run /opt/recipes;
+                  METEOR_EXIT_CODE=$?
+
+                  if [ $METEOR_EXIT_CODE -eq 0 ]; then
+                    {{- if .Values.postCmd.enabled }}
+                    {{ .Values.postCmd.cmd }}
```
Comment (haveiss) [inline stable/meteor/templates/cronjob.yaml:40]: have you test to render it with multiple lines? looks like need indent?
https://helm.sh/docs/chart_template_guide/function_list/#indent

Comment (luthfifahlevi) [reply stable/meteor/templates/cronjob.yaml]: Yes correct mas, thank you for noticing 🙏

**Review by haveiss — APPROVED** [review]

File: stable/meteor/templates/cronjob.yaml [hunk 6]
```diff
@@ -17,140 +17,147 @@ spec:
...
+                - -c
+                - |
+                  meteor run /opt/recipes;
+                  METEOR_EXIT_CODE=$?
+
+                  if [ $METEOR_EXIT_CODE -eq 0 ]; then
+                    {{- if .Values.postCmd.enabled }}
+                    {{ .Values.postCmd.cmd | indent 20 }}
+                    {{- end }}
+                  fi
```
Comment (haveiss) [inline stable/meteor/templates/cronjob.yaml:42]: let's move if higher
```suggestion
                  {{- if .Values.postCmd.enabled }}
                  METEOR_EXIT_CODE=$?

                  if [ $METEOR_EXIT_CODE -eq 0 ]; then
                    {{ .Values.postCmd.cmd | indent 20 }}
                  fi
                  {{- end }}
```

Comment (haveiss) [inline stable/meteor/templates/cronjob.yaml:42]: let's move if higher
```suggestion
                  {{- if .Values.postCmd.enabled }}
                  METEOR_EXIT_CODE=$?

                  if [ $METEOR_EXIT_CODE -eq 0 ]; then
                    {{ .Values.postCmd.cmd | indent 20 }}
                  fi
                  {{- end }}
```

**Review by haveiss — CHANGES_REQUESTED** [review]

File: stable/meteor/templates/cronjob.yaml (hunk 6 above)
Comment (luthfifahlevi) [reply stable/meteor/templates/cronjob.yaml:42]: oh okay mas

File: stable/meteor/templates/cronjob.yaml [hunk 7]
```diff
@@ -169,4 +176,18 @@ spec:
...
+  name: helm-repo-service
+spec:
+  selector:
+    app: helm-repo
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 80
+      nodePort: 30080
+  type: NodePort
```
Comment (haveiss) [inline stable/meteor/templates/cronjob.yaml:193]: what's this?

Comment (luthfifahlevi) [reply stable/meteor/templates/cronjob.yaml]: eh really sorry 🙏  I just try some things on local
reverted mas

File: stable/meteor/templates/cronjob.yaml [hunk 8]
```diff
@@ -17,140 +17,147 @@ spec:
...
+              command:
+                - /bin/sh
+                - -c
+                - |
+                  meteor run /opt/recipes;
+                  {{- if .Values.postCmd.enabled }}
+                  METEOR_EXIT_CODE=$?
+
+                  if [ $METEOR_EXIT_CODE -eq 0 ]; then
+                    {{ .Values.postCmd.cmd | indent 20 }}
```
Comment (haveiss) [inline stable/meteor/templates/cronjob.yaml:40]:
```suggestion
                    {{ .Values.postCmd.cmd | nindent 20 }}
```

File: stable/meteor/values.yaml [hunk 9]
```diff
@@ -38,6 +38,11 @@ resources:
 #         foo: "bar"
 #     sinks:
 #       - name: console
+postCmd:
+  enabled: false
+  cmd:
+    echo "dummy command that will be executed if enabled is true"
```
Comment (haveiss) [inline stable/meteor/values.yaml:44]:
```suggestion
  cmd: ""
#  cmd: |
#    echo "dummy command that will be executed if enabled is true"
```

File: stable/meteor/templates/cronjob.yaml [hunk 10]
```diff
@@ -4,9 +4,9 @@ metadata:
   name: "{{ include "meteor.name" . }}"
   namespace: {{ include "meteor.namespace" . }}
   labels:
-{{- if .Values.labels }}
-{{ toYaml .Values.labels | indent 4 }}
-{{- end }}
+    {{- if .Values.labels }}
+    {{ toYaml .Values.labels | indent 4 }}
+    {{- end }}
```
Comment (haveiss) [inline stable/meteor/templates/cronjob.yaml:9]: sorry,  please fix the whitespaces, it throwing an error

**Review by haveiss — APPROVED** [review]

**Pull request merged**